scripts/tools/EXT_Exp_MetricProgrammable.rst

```diff
@@ -0,0 +1,120 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+    t=tags['$t']
+    T=t.upper()
+%>
+:orphan:
+
+.. _ZET_experimental_programmable_metric:
+
+==========================================
+Metric Programmable Experimental Extension
+==========================================
+
+API
+----
+* Enumerations
+
+    * ${t}_metric_programmable_param_type_exp_t
+
+* Structures
+
+    * ${t}_metric_programmable_exp_properties_t
+    * ${t}_metric_programmable_param_exp_desc_t
+    * ${t}_metric_programmable_param_value_exp_t
+    * ${t}_metric_programmable_param_value_exp_desc_t
+    * ${t}_value_exp_desc_t
+    * ${t}_value_uint64_range_exp_t
+
+* Functions
+
+    * ${t}MetricProgrammableGetExp
+    * ${t}MetricProgrammableGetPropertiesExp
+    * ${t}MetricProgrammableGetParamDescriptionExp
+    * ${t}MetricProgrammableGetParamValueDescriptionExp
+    * ${t}MetricCreateFromProgrammableExp
+    * ${t}MetricGroupCreateExp
+    * ${t}MetricGroupAddMetricExp
+    * ${t}MetricDestroyExp
```

Comment:
we don't have a equivalent MetricCreateExp here. so should this be named differently?

Reply:
The idea is that metrics could be created from different mechansims but destroy to be applicable for all created metrics.
So did not want to have the creation method added to the destroy name.
(Just a note - Destroy would be able to detect how it was created internally)